src/core/lwt.mli

```diff
@@ -21,502 +22,1785 @@
+      completes.
+    + That next I/O happens to be the one that reads [line_1]. {!Lwt_main.run}
+      knows that I/O is supposed to resolve [line_1_promise], so it puts
+      [line_1] into the promise and resolves it.
+    + Resolving is an ordinary OCaml operation. It causes all the callbacks of
+      [line_1_promise] to run, one after another. Each callback is also ordinary
+      OCaml code. In our case, there is only one callback, but in general, there
+      might be several, and they might also resolve additional promises. So,
+      promise resolution triggers a “cascade” of callbacks. Eventually, however,
+      we should run out of callbacks, and control will return to
+      {!Lwt_main.run}.
+    + In our example, our one callback registers a second I/O with
+      {!Lwt_main.run} – the one that will read [line_2]. There are no callbacks
+      left to run after that, so control returns to {!Lwt_main.run}.
+    + {!Lwt_main.run} goes back to sleep again by calling [select(2)], now
+      waiting for the second I/O that we just registered. The loop repeats
+      itself from step 1.
+
+    This has two major implications, one good and one bad. Let's start with the
+    bad one.
+
+    {b (1)} If one of your callbacks enters an infinite loop, calls an
+    Lwt-unfriendly blocking I/O, or just runs for a really long time, it won't
+    return control to {!Lwt_main.run} anytime soon. That means {!Lwt_main.run}
+    won't get a chance to resolve any other Lwt I/O promises, even if the
+    underlying I/O operations complete.
+
+    In case your callback is just using the CPU for a really long time, you can
+    insert a few calls to {!Lwt_main.yield} into it, and resume your computation
+    in callbacks of [yield]. This is basically the same as {!Lwt_unix.sleep 0.}
+    – it's a promise that will be resolved by {!Lwt_main.run} {e after} any
+    other I/O resolutions that are already in its queue.
+
+    {b (2)} The good implication is that all your callbacks run in a single
+    thread. This means that in most situations, you don't have to worry about
+    locks, synchronization, etc. Anything that is in the same callback is
+    guaranteed to run without interruption. Lwt programs are often {e much}
+    easier to write and refactor, than equivalent programs written with threads
+    – but both are concurrent!
+
+
+
+    {3 Guide to the rest of Lwt}
+
+    This module [Lwt] is the pure-OCaml definition of promises and
+    callback-calling. It has a few extras on top of what's described above:
+
+    - {{: #2_Rejection} Rejection}. Lwt promises can actually be resolved in two
+      ways: {e fulfilled} with a value, or {e rejected} with an exception. There
+      is nothing conceptually special about rejection – it's just that you can
+      ask for callbacks to run only on fulfillment, only on rejection, etc.
+    - {{: #2_Cancelation} Cancelation}. This is a special case of rejection,
```

Comment:
perhaps specify that the `Lwt.Canceled` exception is the special case value here.